Modified style guide: func type ctr preceds attr (#1827)
bosagora/agora: doc/Style.md

E.g. a function is: `void foobar ()`, not `void foobar()`.
With parameters: `void foobar (int p1, void* p2)`.
Templated functions are written as `void foobar (T) (T arg)`.
This makes functions easy to `grep` for. Type constructors like `const`, `immutable`,

> `inout`, `shared` preceds other attributes like `pure` or `@nogc`, so
> `void foobar () const @nogc pure` is okay, while `void foobar () pure const @nogc`

Reviewer: ```suggestion
`inout`, `shared` must appear before other attributes like `pure` or `@nogc`, so
`void foobar () const @nogc pure` is correct, while `void foobar () pure const @nogc`
```
